0023-zarf-named-configs/README.md

```diff
@@ -0,0 +1,254 @@
+
+### User Stories (Optional)
+
+#### Story 1
+
+**As** Jacquline **I want** to be able to pre-bake package configuration **so that** I can provide a more declarative package to Ashton.
+
+**Given** I have a Zarf Package created from the below:
+```yaml
+metadata:
+  kind: ZarfPackageConfig
+  name: example
+  version: 0.1.0
+  namespace: example
+
+variables:
+  - name: EXAMPLE
+
+values:
+  - values-default.yaml
+
+components:
+  - name: first
+    ...
+  - name: second
+    ...
+```
+**And** I have a new configuration published from the following
+```yaml
+# option 1
+kind: ZarfNamedConfig
+# option 2
+kind: ZarfDeployConfig
+# option 3
+kind: ZarfConfig
+metadata:
+  name: example-config
+  version: 0.1.0
+
+package:
+  name: example
+  version: 0.1.0
+
+components: [ first ]
+
+namespace: new-namespace
+
+set:
+  EXAMPLE: example
+
+values:
+  - values-override.yaml
+
+adopt-existing-resources: true
+```
+**When** I deploy that package with a `--config` like the below:
+```bash
+zarf package deploy oci://my-registry/example:0.1.0 --config oci://my-registry/example-config:0.1.0
+```
+**Then** Zarf will set the deploy options in accordance with the referenced config
+
+### Risks and Mitigations
+
+This would make it easy to potentially accidentally store secrets in the registry which is not desireable.  We should add documentation about this and potentially prevent the storage of variables that are marked `sensitive` in named configs.
+
+## Design Details
+
+<!-- This is negotiable and would like to hear others' thoughts on a local format for named configs vs having them be OCI-only -->
+Named configs would be created and published through a set of new CLI commands (`zarf config create` and `zarf config publish`).  This would pull together any referenced files or necessary artifacts and either create a local `tar.zst` or publish an OCI reference similar to a package.  This config would then be referenced and applied during a package deployment.
+
+Because package configurations refer to packages a `zarf config list oci://<registry>/<package>` command would also be added that, for an OCI reference, would list the available named configs for that package.
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this proposal.
+
+##### Prerequisite testing updates
+
+NA - This is a modification of existing behavior that should not require prerequisite testing updates.
+
+##### Unit tests
+
+Unit tests would need to be added to ensure that the config was passed into the package deploy library interface correctly.
```

Comment:
What kinds of validation can we run to ensure that the Named Configuration passed in is correct and complete?